packages/xo-server/src/xo-mixins/pool.js

```diff
@@ -81,4 +82,50 @@ export default class Pools {
       await _xo.mergeXenPools(source._xapiId, target._xapiId, force)
     }
   }
+
+  async listPoolsMatchingCriteria({
+    minAvailableHostMemory = 0,
+    minAvailableSrSize = 0,
+    minHostCpus = 0,
+    minHostVersion,
+    poolNameRegExp,
+    srNameRegExp,
+  }) {
+    const hostsByPool = {}
+    const srsByPool = {}
+    const pools = []
+    for (const obj of this._xo.objects.values()) {
+      if (obj.type === 'host') {
+        if (hostsByPool[obj.$pool] === undefined) {
+          hostsByPool[obj.$pool] = []
+        }
+        hostsByPool[obj.$pool].push(obj)
+      } else if (obj.type === 'SR') {
+        if (srsByPool[obj.$pool] === undefined) {
+          srsByPool[obj.$pool] = []
+        }
+        srsByPool[obj.$pool].push(obj)
+      } else if (obj.type === 'pool') {
+        pools.push(obj)
+      } else {
+        continue
+      }
+    }
+
+    return pools.filter(
+      pool =>
+        (poolNameRegExp === undefined || new RegExp(poolNameRegExp).test(pool.name_label)) &&
```

Comment:
Create the regexp only once.

Also, you can simplify the code by creating a dedicated function:

```js
const checkPoolName = poolNameRegExp === undefined
  ? lodash.stubTrue
  : RegExp.prototype.test.bind(new RegExp(poolNameRegExp))
```

And do the same for srNameRefExp.